frontend/app.js

```diff
@@ -0,0 +1,510 @@
+      taskName: '',
+      taskDescription: '',
+      childVisible: false,
+      submitBar: false,
+      completedState: 0
+    };
+  }
+
+  componentDidMount() {
+    fetch('/task')
+      .then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  addTask(event) {
+    event.preventDefault();
+    var name = this.state.taskName
+    var description = this.state.taskDescription
+    if (name == '') {
+      alert('you must write something')
+    } else {
+
+      fetch('/task/', {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: name,
+          description: description
+        })
+      }).then((data) => {
+          return data.json()
+        })
+        .then((json) => {
+
+          this.setState({
+            childVisible: false,
+            tasks: json.tasks,
+            taskName: '',
+            taskDescription: '',
+            submitBar: false
+          })
+
+      })
+    }
+  }
+
+  removeTask(event) {
+    fetch('/task/', {
+      method: 'DELETE',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        id: event
+      })
+    }).then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  completeTask(event) {
+    console.log(event)
+    fetch('/task/', {
+      method: 'PUT',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        id: event
+      })
+    }).then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  onTaskChange(event) {
+    this.setState({
+      taskName: event.target.value,
+    });
+    if (event.target.value.length > 0) {
+      this.setState({
+        childVisible: true
+      })
+    } else {
+      this.setState({
+        submitBar: false,
+        childVisible: false,
+        taskDescription: ''
+      })
+    }
+  }
+
+  onDesciptionChange(event) {
+    this.setState({
+      taskDescription: event.target.value
+    });
+    if (event.target.value.length > 0) {
+      this.setState({
+        submitBar: true
+      })
+    } else {
+
+      this.setState({
+        submitBar: false,
+
+      })
+    }
+  }
+
+  changeTab(event) {
+    if (this.state.completedState === 1) {
+      this.setState({
+        completedState: 0
+      })
+    } else if (this.state.completedState === 0) {
+      this.setState({
+        completedState: 1
+      })
+    }
+  }
+
+    render() {
+      if (this.state.tasks && this.state.completedState === 0) {
```

Comment:
Your control logic is a little repetitive here, and I think that's because you didn't realize you could do [conditionals in JSX](https://facebook.github.io/react/docs/conditional-rendering.html). 

```jsx
{this.state.tasks && this.state.completedState === 0 && (
  <Blah />
)}
{this.state.tasks && this.state.completedState === 1 && (
  <SomeComponent />
)}
<Todo ... />
```